File: week-9/protoseo/소수 찾기.java
import java.util.*;

class Solution {
    Set<Integer> candidate = new TreeSet<>();
    char[] numbersToArray;
    public int solution(String numbers) {
        numbersToArray = numbers.toCharArray();
        find(0, "0", new boolean[numbers.length()]);

        int answer = 0;
        for (int c : candidate) {
            if (isPrime(c)) {
                answer++;
            }
        }
        return answer;
    }

    void find(int idx, String s, boolean[] isVisited) {
        candidate.add(Integer.parseInt(s));
        if (idx == numbersToArray.length) {
            return;
        }
        for (int i = 0; i < numbersToArray.length; i++) {
            if (isVisited[i]) {
                continue;
            }
            isVisited[i] = true;
            find(idx + 1, s + numbersToArray[i], isVisited);
            isVisited[i] = false;
        }
    }

    boolean isPrime(int num) {
        if (num <= 1) {
            return false;
        }
        for (int i = 2; i * i <= num; i++) {
            if (num % i == 0) {
                return false;
            }
        }
        return true;
    }
}

File: week-09/dahyen0o/소수-찾기.py
import math
from itertools import permutations

def solution(numbers):
    answer = set()
    for counts in range(1, len(numbers) + 1):
        for case in permutations(numbers, counts):
            num = int(''.join(case))
            if is_prime(num):
                answer.add(num)
    return len(answer)

def is_prime(num):
    if num == 0 or num == 1:
        return False
    
    for i in range(2, int(math.sqrt(num)) + 1):
        if num % i == 0:
            return False
    return True
